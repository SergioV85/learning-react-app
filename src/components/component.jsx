import React from 'react';

export class Subtitle extends React.Component {
  render() {
    return <h2>This is subtitle, simple component</h2>;
  }
}