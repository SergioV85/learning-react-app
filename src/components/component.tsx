import * as React from 'react';

export class Subtitle extends React.Component {
  public render() {
    return <h2 className="custom-header">This is subtitle, simple component</h2>;
  }
}