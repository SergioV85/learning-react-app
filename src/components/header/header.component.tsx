import * as React from 'react';

interface IHeaderProps {
  showBackButton?: boolean;
  onBackToList(): void;
}
// tslint:disable:max-line-length
export const Header = (props: IHeaderProps) =>
  <div className='c-header'>
    <span className='c-header__logo'>netflixroulette</span>
    {props.showBackButton ? <button className='btn c-header__back-button' onClick={props.onBackToList}> Search </button> : null}
  </div>;
// tslint:enable:max-line-length
