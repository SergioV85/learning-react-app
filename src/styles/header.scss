@import './variables';

.c-header {
  align-items: center;
  display: flex;
  justify-content: space-between;

  &__logo {
    color: $primary-red;
    font-family: $tilte-font;
    font-size: 1rem;
    font-weight: 700;
  }

  &__back-button {
    color: $primary-red;
    font-family: $main-font;
    font-size: 0.8rem;
  }
}