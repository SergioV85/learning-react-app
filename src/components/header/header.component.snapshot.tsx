import { shallow } from 'enzyme';
import * as React from 'react';
import { Header } from './header.component';

describe('Header Component Snapshot', () => {
  const mockedBackToList = jest.fn();

  it('should renders correctly', () => {
    const headerSnapshot = shallow(
      <Header showBackButton={true} onBackToList={mockedBackToList} />,
    );
    expect(headerSnapshot)
      .toMatchSnapshot();
  });
});
