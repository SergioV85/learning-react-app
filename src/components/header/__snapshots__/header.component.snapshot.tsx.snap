// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Snapshot should renders correctly 1`] = `
<div
  className="c-header"
>
  <span
    className="c-header__logo"
  >
    netflixroulette
  </span>
  <button
    className="btn c-header__back-button"
    onClick={[MockFunction]}
  >
     Search 
  </button>
</div>
`;
